{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "661ae8e3",
   "metadata": {},
   "source": [
    "### Python Code Review\n",
    "- Overview of code and its application (eLife paper)\n",
    "- nodes.csv; Edits to current for loop, and converting over 3 other loops to be more 'DRY' <br>One loop possibly becoming a function\n",
    "- Folder conversion"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a6afe653",
   "metadata": {},
   "source": [
    "### To Do (in order of when they show up)\n",
    "\n",
    "- Warning (reading mike's CSVs)\n",
    "- NaN: qualifier and inner1\n",
    "- Make time stamp into a for loop\n",
    "- Why are categories much less than for WDumper and wdsub? \n",
    "- Why is 'Chemical Hazard' empty?\n",
    "- NaN in the 'name' section of some CSVs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d914b2c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "# comparison of no grep\n",
    "import pandas as pd\n",
    "\n",
    "import time \n",
    "from datetime import datetime\n",
    "\n",
    "import os"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "aa9f66ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/sulhasan/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (3,4) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "nodes_mike = pd.read_csv('file-examples/nodes_2019-09-03.csv')  \n",
    "edges_mike = pd.read_csv('file-examples/edges_2019-09-03.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "25004f3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:ID</th>\n",
       "      <th>:LABEL</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q4735440</td>\n",
       "      <td>Compound</td>\n",
       "      <td>alpidem</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q5983577</td>\n",
       "      <td>Compound</td>\n",
       "      <td>ibacitabine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q221174</td>\n",
       "      <td>Compound</td>\n",
       "      <td>topiramate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q2016367</td>\n",
       "      <td>Compound</td>\n",
       "      <td>tetraphenylborate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q168685</td>\n",
       "      <td>Compound</td>\n",
       "      <td>fentin acetate</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        :ID    :LABEL               name\n",
       "0  Q4735440  Compound            alpidem\n",
       "1  Q5983577  Compound        ibacitabine\n",
       "2   Q221174  Compound         topiramate\n",
       "3  Q2016367  Compound  tetraphenylborate\n",
       "4   Q168685  Compound     fentin acetate"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_mike.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e391fb37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:END_ID</th>\n",
       "      <th>:TYPE</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>inner1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q144085</td>\n",
       "      <td>Q412496</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q3712521</td>\n",
       "      <td>Q419394</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q410061</td>\n",
       "      <td>Q417597</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q174259</td>\n",
       "      <td>Q7739</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q3791612</td>\n",
       "      <td>Q41576</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  :START_ID  :END_ID                              :TYPE qualifier inner1\n",
       "0   Q144085  Q412496  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "1  Q3712521  Q419394  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "2   Q410061  Q417597  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "3   Q174259    Q7739  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "4  Q3791612   Q41576  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges_mike.head() # qualifier NaN (both head and tail)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e7774e48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:END_ID</th>\n",
       "      <th>:TYPE</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>inner1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>614760</th>\n",
       "      <td>Q21112306</td>\n",
       "      <td>Q14878358</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14761970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614761</th>\n",
       "      <td>Q21111886</td>\n",
       "      <td>Q14864067</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14863719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614762</th>\n",
       "      <td>Q11705457</td>\n",
       "      <td>Q14864067</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14863719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614763</th>\n",
       "      <td>Q423701</td>\n",
       "      <td>Q14878358</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14761970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614764</th>\n",
       "      <td>Q3616234</td>\n",
       "      <td>Q14878358</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14761970</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        :START_ID    :END_ID              :TYPE qualifier     inner1\n",
       "614760  Q21112306  Q14878358  UP_REGULATES_GuBP       NaN  Q14761970\n",
       "614761  Q21111886  Q14864067  UP_REGULATES_GuBP       NaN  Q14863719\n",
       "614762  Q11705457  Q14864067  UP_REGULATES_GuBP       NaN  Q14863719\n",
       "614763    Q423701  Q14878358  UP_REGULATES_GuBP       NaN  Q14761970\n",
       "614764   Q3616234  Q14878358  UP_REGULATES_GuBP       NaN  Q14761970"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges_mike.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "d3688b0b",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0008685986200968424 for activesite.ndjson\n"
     ]
    }
   ],
   "source": [
    "# Time stamp\n",
    "\n",
    "timeStringNow = datetime.now().strftime(\"+%Y-%m-%dT00:00:00Z\") \n",
    "\n",
    "start_time = time.time()\n",
    "activesite = pd.read_json('files/activesite.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for activesite.ndjson\") \n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0a3fc565",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0002136826515197754 for activesite.ndjson\n",
      "0.0028772830963134767 for anatomy.ndjson\n",
      "0.0011151631673177083 for biologicalpathway.ndjson\n",
      "0.01685828367869059 for biologicalprocess.ndjson\n",
      "0.00015334685643513996 for bindingsite.ndjson\n",
      "0.004989516735076904 for cellularcomponent.ndjson\n",
      "4.881223042805989e-05 for chemicalhazard.ndjson\n",
      "0.2749996503194173 for compound.ndjson\n",
      "0.00136488676071167 for disease.ndjson\n",
      "10.554157332579296 for gene.ndjson\n",
      "0.005231932799021403 for medicalspecialty.ndjson\n",
      "0.05962743361790975 for molecularfunction.ndjson\n",
      "63.00341316461563 for protein.ndjson\n",
      "0.005757097403208415 for proteindomain.ndjson\n",
      "0.0073135972023010256 for proteinfamily.ndjson\n",
      "0.0003326535224914551 for sequencevariant.ndjson\n",
      "0.00011376539866129558 for structuralmotif.ndjson\n",
      "0.00019429922103881837 for symptom.ndjson\n"
     ]
    }
   ],
   "source": [
    "# Create time stamp\n",
    "timeStringNow = datetime.now().strftime(\"+%Y-%m-%dT00:00:00Z\") \n",
    "\n",
    "start_time = time.time()\n",
    "activesite = pd.read_json('files/activesite.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for activesite.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "anatomy = pd.read_json('files/anatomy.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for anatomy.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "biologicalpathway = pd.read_json('files/biologicalpathway.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for biologicalpathway.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "biologicalprocess = pd.read_json('files/biologicalprocess.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for biologicalprocess.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "bindingsite = pd.read_json('files/bindingsite.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for bindingsite.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "cellularcomponent = pd.read_json('files/cellularcomponent.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for cellularcomponent.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "chemicalhazard = pd.read_json('files/chemicalhazard.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for chemicalhazard.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "compound = pd.read_json('files/compound.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for compound.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "disease = pd.read_json('files/disease.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for disease.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "gene = pd.read_json('files/gene.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for gene.ndjson\") # 2 min\n",
    "\n",
    "start_time = time.time()\n",
    "medicalspecialty = pd.read_json('files/medicalspecialty.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for medicalspecialty.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "molecularfunction = pd.read_json('files/molecularfunction.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for molecularfunction.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "protein = pd.read_json('files/protein.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for protein.ndjson\") # 12 min\n",
    "\n",
    "start_time = time.time()\n",
    "proteindomain = pd.read_json('files/proteindomain.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for proteindomain.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "proteinfamily = pd.read_json('files/proteinfamily.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for proteinfamily.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "sequencevariant = pd.read_json('files/sequencevariant.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for sequencevariant.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "structuralmotif = pd.read_json('files/structuralmotif.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for structuralmotif.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "symptom = pd.read_json('files/symptom.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for symptom.ndjson\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "f9be814d",
   "metadata": {},
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (<ipython-input-6-cf5d1c3a1ac1>, line 13)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-6-cf5d1c3a1ac1>\"\u001b[0;36m, line \u001b[0;32m13\u001b[0m\n\u001b[0;31m    node_name = [*.ndjson]\u001b[0m\n\u001b[0m                  ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "directory = 'files-test'\n",
    "    \n",
    "node_dict = []    \n",
    "    \n",
    "for filename in os.listdir(directory):\n",
    "    \n",
    "    f = os.path.join(directory, filename)\n",
    "    timeStringNow = datetime.now().strftime(\"+%Y-%m-%dT00:00:00Z\") \n",
    "    \n",
    "    if os.path.isfile(f):\n",
    "        start_time = time.time()\n",
    "        \n",
    "        node_name = [*.ndjson]\n",
    "        node_dict[node_name] = pd.read_json(f, lines=True)\n",
    "\n",
    "        end_time = time.time()\n",
    "        print((end_time - start_time)/60, \"for\", f) # How do I remove the 'files/' ?\n",
    "   \n",
    "\n",
    "activesite = pd.read_json('files/activesite.ndjson', lines=True)  \n",
    "\n",
    "# create a dictionary with the keys that have part of the file name   \n",
    "# values be data\n",
    "\n",
    "content.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "9d340508",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ".ndjson\n",
      "\n"
     ]
    }
   ],
   "source": [
    "filename = \"/a/b/c/d/.ndjson\"\n",
    "print(os.path.basename(filename))\n",
    "print(os.path.basename(filename).strip(\".ndjson\"))\n",
    "\n",
    "node_dict={}\n",
    "node_dict[node_name]=\"my pandas object\"\n",
    "for key, values in node_dict.items(\n",
    "print(key,values))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "e2f3c566",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biological Process           24718\n",
       "Gene                         23737\n",
       "Protein Family               10406\n",
       "Compound                      8557\n",
       "Disease                       8302\n",
       "Protein Domain                5230\n",
       "Molecular Function            4397\n",
       "Biological Pathway            2461\n",
       "Sequence Variant              1835\n",
       "Cellular Component            1785\n",
       "Chemical Hazard                687\n",
       "Super-Secondary Structure      663\n",
       "Chemical Role                  499\n",
       "Symptom                        228\n",
       "Anatomical Structure           218\n",
       "Active Site                    114\n",
       "Structural Motif               109\n",
       "Medical Specialty               69\n",
       "Binding Site                    69\n",
       "Name: :LABEL, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_mike[':LABEL'].value_counts() # 19 categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "eb9c03b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Biological Process \t 3562\n",
      "Gene \t\t\t 89348\n",
      "Protein Family \t\t 2441\n",
      "Compound \t\t 18718\n",
      "Disease \t\t 337\n",
      "Protein Domain \t\t 1117\n",
      "Molecular Function \t 1307\n",
      "Biological Pathway \t 52\n",
      "Sequence Variant \t 80\n",
      "Cellular Component \t 511\n",
      "Chemical Hazard \t 0\n",
      "Anatomical Structure \t 368\n",
      "Active Site \t\t 9\n",
      "Structural Motif \t 35\n",
      "Binding Site \t\t 10\n",
      "Medical Specialty \t 24\n",
      "\n",
      "Protein \t\t 63604\n"
     ]
    }
   ],
   "source": [
    "# Convert to for loop\n",
    "\n",
    "print( \"Biological Process \\t\", len(biologicalprocess)) \n",
    "print( \"Gene \\t\\t\\t\", len(gene)) \n",
    "print( \"Protein Family \\t\\t\", len(proteinfamily)) \n",
    "print( \"Compound \\t\\t\", len(compound))\n",
    "print( \"Disease \\t\\t\", len(disease))\n",
    "print( \"Protein Domain \\t\\t\", len(proteindomain))\n",
    "print( \"Molecular Function \\t\", len(molecularfunction))\n",
    "print( \"Biological Pathway \\t\", len(biologicalpathway))\n",
    "print( \"Sequence Variant \\t\", len(sequencevariant))\n",
    "print( \"Cellular Component \\t\", len(cellularcomponent))\n",
    "print( \"Chemical Hazard \\t\", len(chemicalhazard))\n",
    "print( \"Anatomical Structure \\t\", len(anatomy))\n",
    "print( \"Active Site \\t\\t\", len(activesite))\n",
    "print( \"Structural Motif \\t\", len(structuralmotif))\n",
    "print( \"Binding Site \\t\\t\", len(bindingsite))\n",
    "print( \"Medical Specialty \\t\", len(medicalspecialty))\n",
    "print( \"\\nProtein \\t\\t\", len(protein)) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "473f934a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0f38f5d5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "         :ID      ,:LABEL                 ,name\n",
      "0  Q24721060  Active Site  InterPro Active Site\n",
      "1  Q24721301  Active Site  InterPro Active Site\n",
      "2  Q24721995  Active Site  InterPro Active Site\n",
      "     :ID               ,:LABEL  \\\n",
      "0  Q7365  Anatomical Structure   \n",
      "1  Q9377  Anatomical Structure   \n",
      "2  Q9609  Anatomical Structure   \n",
      "\n",
      "                                               ,name  \n",
      "0                 contractile soft tissue of mammals  \n",
      "1  internal organ in most animals, including vert...  \n",
      "2  blood vessel that carry deoxygenated blood tow...  \n",
      "         :ID             ,:LABEL  \\\n",
      "0  Q21694734  Biological Pathway   \n",
      "1  Q28031274  Biological Pathway   \n",
      "2  Q29891990  Biological Pathway   \n",
      "\n",
      "                                               ,name  \n",
      "0                                  metabolic pathway  \n",
      "1  Glycolysis is the metabolic pathway that conve...  \n",
      "2  The mitogen-activated protein kinase (MAPK) ca...  \n",
      "       :ID             ,:LABEL  \\\n",
      "0    Q1057  Biological Process   \n",
      "1   Q99974  Biological Process   \n",
      "2  Q134624  Biological Process   \n",
      "\n",
      "                                               ,name  \n",
      "0  set of chemical reactions and pathways by whic...  \n",
      "1  symbiotic association between a fungus and the...  \n",
      "2  biological pollination process occurring in pl...  \n",
      "         :ID       ,:LABEL                  ,name\n",
      "0  Q24722105  Binding Site  InterPro Binding Site\n",
      "1  Q24725852  Binding Site  InterPro Binding Site\n",
      "2  Q24725899  Binding Site  InterPro Binding Site\n",
      "      :ID             ,:LABEL  \\\n",
      "0  Q29548  Cellular Component   \n",
      "1  Q30869  Cellular Component   \n",
      "2  Q61333  Cellular Component   \n",
      "\n",
      "                                               ,name  \n",
      "0  biological membrane separating the interior of...  \n",
      "1  largest structure in the nucleus of eukaryotic...  \n",
      "2  The sex chromosome present in both sexes of sp...  \n",
      "Empty DataFrame\n",
      "Columns: [:ID, ,:LABEL, ,name]\n",
      "Index: []\n",
      "      :ID   ,:LABEL               ,name\n",
      "0   Q2546  Compound   chemical compound\n",
      "1  Q19052  Compound  inorganic compound\n",
      "2  Q32703  Compound   chemical compound\n",
      "       :ID  ,:LABEL                                  ,name\n",
      "0   Q68854  Disease   injury caused by a bite from a snake\n",
      "1  Q119280  Disease  congenital disorder of urinary system\n",
      "2  Q143790  Disease              neuropsychiatric disorder\n",
      "       :ID ,:LABEL                                            ,name\n",
      "0   Q30555    Gene                 gene that can become an oncogene\n",
      "1  Q286987    Gene  protein-coding gene in the species Homo sapiens\n",
      "2  Q390540    Gene  protein-coding gene in the species Homo sapiens\n",
      "       :ID            ,:LABEL  \\\n",
      "0  Q120569  Medical Specialty   \n",
      "1  Q123028  Medical Specialty   \n",
      "2  Q162555  Medical Specialty   \n",
      "\n",
      "                                               ,name  \n",
      "0  branch of medicine focused on the digestive sy...  \n",
      "1  field of medicine dealing with the care of chi...  \n",
      "2             branch of medicine dealing with cancer  \n",
      "         :ID             ,:LABEL  \\\n",
      "0   Q1012651  Molecular Function   \n",
      "1  Q14358843  Molecular Function   \n",
      "2  Q14599285  Molecular Function   \n",
      "\n",
      "                                               ,name  \n",
      "0  Catalysis of the endonucleolytic cleavage of R...  \n",
      "1  Catalysis of an oxidation-reduction (redox) re...  \n",
      "2  Catalysis of an oxidation-reduction (redox) re...  \n",
      "       :ID  ,:LABEL                                    ,name\n",
      "0   Q63398  Protein  mammalian protein found in Homo sapiens\n",
      "1  Q171604  Protein  mammalian protein found in Homo sapiens\n",
      "2  Q205042  Protein  mammalian protein found in Homo sapiens\n",
      "        :ID         ,:LABEL            ,name\n",
      "0  Q1829843  Protein Domain  InterPro Domain\n",
      "1  Q5009288  Protein Domain  InterPro Domain\n",
      "2  Q6718295  Protein Domain              NaN\n",
      "       :ID         ,:LABEL                                              ,name\n",
      "0  Q408477  Protein Family                                    InterPro family\n",
      "1  Q409145  Protein Family  group of calcium sensors and calcium signal mo...\n",
      "2  Q410959  Protein Family                                    InterPro Family\n",
      "         :ID           ,:LABEL            ,name\n",
      "0  Q28371440  Sequence Variant  genetic variant\n",
      "1  Q28444966  Sequence Variant  genetic variant\n",
      "2  Q28445004  Sequence Variant  genetic variant\n",
      "         :ID           ,:LABEL ,name\n",
      "0   Q2007385  Structural Motif   NaN\n",
      "1   Q5445279  Structural Motif   NaN\n",
      "2  Q19903500  Structural Motif   NaN\n",
      "       :ID  ,:LABEL                                              ,name\n",
      "0  Q188611  Symptom  mood state characterized by persistent and per...\n",
      "1  Q653197  Symptom                                     skin condition\n",
      "2  Q936070  Symptom                   medical problem of jaw clenching\n"
     ]
    }
   ],
   "source": [
    "# Write as function and create into for loop?\n",
    "\n",
    "with open('activesite.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in activesite.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Active Site#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Active Site#\", file = file)\n",
    "activesitefinal = pd.read_csv('activesite.csv', sep=\"#\")\n",
    "\n",
    "with open('anatomy.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in anatomy.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Anatomical Structure#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Anatomical Structure#\", file = file)\n",
    "anatomyfinal = pd.read_csv('anatomy.csv', sep=\"#\")\n",
    "\n",
    "with open('biologicalpathway.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in biologicalpathway.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Biological Pathway#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Biological Pathway#\", file = file)\n",
    "biologicalpathwayfinal = pd.read_csv('biologicalpathway.csv', sep=\"#\")\n",
    "\n",
    "with open('biologicalprocess.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in biologicalprocess.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Biological Process#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Biological Process#\", file = file)\n",
    "biologicalprocessfinal = pd.read_csv('biologicalprocess.csv', sep=\"#\")\n",
    "\n",
    "with open('bindingsite.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in bindingsite.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Binding Site#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Binding Site#\", file = file)\n",
    "bindingsitefinal = pd.read_csv('bindingsite.csv', sep=\"#\")\n",
    "\n",
    "with open('cellularcomponent.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in cellularcomponent.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Cellular Component#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Cellular Component#\", file = file)\n",
    "cellularcomponentfinal = pd.read_csv('cellularcomponent.csv', sep=\"#\")\n",
    "\n",
    "with open('chemicalhazard.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in chemicalhazard.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Chemical Hazard#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Chemical Hazard#\", file = file)\n",
    "chemicalhazardfinal = pd.read_csv('chemicalhazard.csv', sep=\"#\")\n",
    "\n",
    "with open('compound.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in compound.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Compound#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Compound#\", file = file)\n",
    "compoundfinal = pd.read_csv('compound.csv', sep=\"#\")\n",
    "\n",
    "with open('disease.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in disease.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Disease#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Disease#\", file = file)\n",
    "diseasefinal = pd.read_csv('disease.csv', sep=\"#\")\n",
    "\n",
    "with open('gene.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in gene.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Gene#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Gene#\", file = file)\n",
    "genefinal = pd.read_csv('gene.csv', sep=\"#\")\n",
    "\n",
    "with open('medicalspecialty.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in medicalspecialty.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Medical Specialty#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Medical Specialty#\", file = file)\n",
    "medicalspecialtyfinal = pd.read_csv('medicalspecialty.csv', sep=\"#\")\n",
    "\n",
    "with open('molecularfunction.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in molecularfunction.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Molecular Function#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Molecular Function#\", file = file)\n",
    "molecularfunctionfinal = pd.read_csv('molecularfunction.csv', sep=\"#\")\n",
    "\n",
    "with open('protein.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in protein.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Protein#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Protein#\", file = file)\n",
    "proteinfinal = pd.read_csv('protein.csv', sep=\"#\")\n",
    "\n",
    "with open('proteindomain.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in proteindomain.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Protein Domain#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Protein Domain#\", file = file)\n",
    "proteindomainfinal = pd.read_csv('proteindomain.csv', sep=\"#\")\n",
    "\n",
    "with open('proteinfamily.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in proteinfamily.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Protein Family#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Protein Family#\", file = file)\n",
    "proteinfamilyfinal = pd.read_csv('proteinfamily.csv', sep=\"#\")\n",
    "\n",
    "with open('sequencevariant.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in sequencevariant.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Sequence Variant#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Sequence Variant#\", file = file)\n",
    "sequencevariantfinal = pd.read_csv('sequencevariant.csv', sep=\"#\")\n",
    "\n",
    "with open('structuralmotif.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in structuralmotif.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Structural Motif#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Structural Motif#\", file = file)\n",
    "structuralmotiffinal = pd.read_csv('structuralmotif.csv', sep=\"#\")\n",
    "\n",
    "with open('symptom.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID#', ':LABEL#', 'name'])\n",
    "    for index, row in symptom.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Symptom#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Symptom#\", file = file)\n",
    "symptomfinal = pd.read_csv('symptom.csv', sep=\"#\")\n",
    "\n",
    "print(activesitefinal.head(3))\n",
    "print(anatomyfinal.head(3))\n",
    "print(biologicalpathwayfinal.head(3))\n",
    "print(biologicalprocessfinal.head(3))\n",
    "print(bindingsitefinal.head(3))\n",
    "print(cellularcomponentfinal.head(3))\n",
    "print(chemicalhazardfinal.head(3))\n",
    "print(compoundfinal.head(3))\n",
    "print(diseasefinal.head(3))\n",
    "print(genefinal.head(3))\n",
    "print(medicalspecialtyfinal.head(3))\n",
    "print(molecularfunctionfinal.head(3))\n",
    "print(proteinfinal.head(3))\n",
    "print(proteindomainfinal.head(3))\n",
    "print(proteinfamilyfinal.head(3))\n",
    "print(sequencevariantfinal.head(3))\n",
    "print(structuralmotiffinal.head(3))\n",
    "print(symptomfinal.head(3))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "d10b913b",
   "metadata": {},
   "outputs": [],
   "source": [
    "# convert into for loop\n",
    "\n",
    "# make sure same number of columns (1:3)\n",
    "activesitefinal = activesitefinal[[':ID', ',:LABEL', ',name']]\n",
    "anatomyfinal = anatomyfinal[[':ID', ',:LABEL', ',name']]\n",
    "biologicalpathwayfinal = biologicalpathwayfinal[[':ID', ',:LABEL', ',name']]\n",
    "biologicalprocessfinal = biologicalprocessfinal[[':ID', ',:LABEL', ',name']]\n",
    "bindingsitefinal = bindingsitefinal[[':ID', ',:LABEL', ',name']]\n",
    "cellularcomponentfinal = cellularcomponentfinal[[':ID', ',:LABEL', ',name']]\n",
    "chemicalhazardfinal = chemicalhazardfinal[[':ID', ',:LABEL', ',name']]\n",
    "compoundfinal = compoundfinal[[':ID', ',:LABEL', ',name']]\n",
    "diseasefinal = diseasefinal[[':ID', ',:LABEL', ',name']]\n",
    "genefinal = genefinal[[':ID', ',:LABEL', ',name']]\n",
    "medicalspecialtyfinal = medicalspecialtyfinal[[':ID', ',:LABEL', ',name']]\n",
    "molecularfunctionfinal = molecularfunctionfinal[[':ID', ',:LABEL', ',name']]\n",
    "proteinfinal = proteinfinal[[':ID', ',:LABEL', ',name']]\n",
    "proteindomainfinal = proteindomainfinal[[':ID', ',:LABEL', ',name']]\n",
    "proteinfamilyfinal = proteinfamilyfinal[[':ID', ',:LABEL', ',name']]\n",
    "sequencevariantfinal = sequencevariantfinal[[':ID', ',:LABEL', ',name']]\n",
    "structuralmotiffinal = structuralmotiffinal[[':ID', ',:LABEL', ',name']]\n",
    "symptomfinal = symptomfinal[[':ID', ',:LABEL', ',name']]\n",
    "\n",
    "nodes = pd.concat([activesitefinal, anatomyfinal, biologicalpathwayfinal, \n",
    "                   biologicalprocessfinal, bindingsitefinal, cellularcomponentfinal,\n",
    "                   chemicalhazardfinal, compoundfinal, diseasefinal, \n",
    "                   genefinal, medicalspecialtyfinal, molecularfunctionfinal, \n",
    "                   proteinfinal, proteindomainfinal, proteinfamilyfinal, \n",
    "                   sequencevariantfinal, structuralmotiffinal, symptomfinal], axis=0)\n",
    "nodes.to_csv('nodes.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "4ffe41a3",
   "metadata": {},
   "outputs": [],
   "source": [
    "# files.ndjson\n",
    "# files.csv\n",
    "\n",
    "# 1 [this approach]\n",
    "## Move all *.csv files => files-new\n",
    "## setwd files-new => concat to nodes.csv\n",
    "\n",
    "# Does directory exist? If not, make directory\n",
    "# Then, move all files to new folder\n",
    "\n",
    "# 2\n",
    "## Move all *.csv files EXCEPT for nodes.csv => files-new\n",
    "\n",
    "if not os.path.exists(\"files-new\"):\n",
    "    os.mkdir(\"files-new\")\n",
    "    \n",
    "for  filename in os.listdir('.')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "7c4233d4",
   "metadata": {},
   "outputs": [],
   "source": [
    "# move all with .csv (that's not nodes.csv) to another folder within\n",
    "\n",
    "## import os\n",
    "## import shutil\n",
    "\n",
    "### src_dir = \"/Users/sulhasan/Desktop/Project/WRP/src/Jan3-2022\"\n",
    "### dst_dir = \"/Users/sulhasan/Desktop/Project/WRP/src/Jan3-2022/files-new\"\n",
    "\n",
    "### for root, dirs, files in os.walk(src_dir):\n",
    "#    for f in files:\n",
    "#        if f.endswith('.csv'):\n",
    "#            shutil.copy(os.path.join(root,f), dst_dir)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
