{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "a6afe653",
   "metadata": {},
   "source": [
    "### To Do (in order of when they show up)\n",
    "\n",
    "- Warning (reading mike's CSVs)\n",
    "- NaN: qualifier and inner1\n",
    "- Make time stamp into a for loop\n",
    "- Is compound category less time because of medical specialty? \n",
    "- Double-check identity for head of each category\n",
    "- Why is 'Chemical Hazard' empty?"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d914b2c8",
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "import time \n",
    "from datetime import datetime"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "aa9f66ad",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/Users/sulhasan/opt/anaconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3165: DtypeWarning: Columns (3,4) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "nodes_mike = pd.read_csv('file-examples/nodes_2019-09-03.csv')  \n",
    "edges_mike = pd.read_csv('file-examples/edges_2019-09-03.csv')  "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "25004f3c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:ID</th>\n",
       "      <th>:LABEL</th>\n",
       "      <th>name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q4735440</td>\n",
       "      <td>Compound</td>\n",
       "      <td>alpidem</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q5983577</td>\n",
       "      <td>Compound</td>\n",
       "      <td>ibacitabine</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q221174</td>\n",
       "      <td>Compound</td>\n",
       "      <td>topiramate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q2016367</td>\n",
       "      <td>Compound</td>\n",
       "      <td>tetraphenylborate</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q168685</td>\n",
       "      <td>Compound</td>\n",
       "      <td>fentin acetate</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        :ID    :LABEL               name\n",
       "0  Q4735440  Compound            alpidem\n",
       "1  Q5983577  Compound        ibacitabine\n",
       "2   Q221174  Compound         topiramate\n",
       "3  Q2016367  Compound  tetraphenylborate\n",
       "4   Q168685  Compound     fentin acetate"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_mike.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e391fb37",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:END_ID</th>\n",
       "      <th>:TYPE</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>inner1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Q144085</td>\n",
       "      <td>Q412496</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Q3712521</td>\n",
       "      <td>Q419394</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Q410061</td>\n",
       "      <td>Q417597</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Q174259</td>\n",
       "      <td>Q7739</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Q3791612</td>\n",
       "      <td>Q41576</td>\n",
       "      <td>SIGNIFICANT_DRUG_INTERACTION_CdiC</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  :START_ID  :END_ID                              :TYPE qualifier inner1\n",
       "0   Q144085  Q412496  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "1  Q3712521  Q419394  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "2   Q410061  Q417597  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "3   Q174259    Q7739  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN\n",
       "4  Q3791612   Q41576  SIGNIFICANT_DRUG_INTERACTION_CdiC       NaN    NaN"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges_mike.head() # qualifier NaN (both head and tail)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e7774e48",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>:START_ID</th>\n",
       "      <th>:END_ID</th>\n",
       "      <th>:TYPE</th>\n",
       "      <th>qualifier</th>\n",
       "      <th>inner1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>614760</th>\n",
       "      <td>Q21112306</td>\n",
       "      <td>Q14878358</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14761970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614761</th>\n",
       "      <td>Q21111886</td>\n",
       "      <td>Q14864067</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14863719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614762</th>\n",
       "      <td>Q11705457</td>\n",
       "      <td>Q14864067</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14863719</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614763</th>\n",
       "      <td>Q423701</td>\n",
       "      <td>Q14878358</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14761970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>614764</th>\n",
       "      <td>Q3616234</td>\n",
       "      <td>Q14878358</td>\n",
       "      <td>UP_REGULATES_GuBP</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Q14761970</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        :START_ID    :END_ID              :TYPE qualifier     inner1\n",
       "614760  Q21112306  Q14878358  UP_REGULATES_GuBP       NaN  Q14761970\n",
       "614761  Q21111886  Q14864067  UP_REGULATES_GuBP       NaN  Q14863719\n",
       "614762  Q11705457  Q14864067  UP_REGULATES_GuBP       NaN  Q14863719\n",
       "614763    Q423701  Q14878358  UP_REGULATES_GuBP       NaN  Q14761970\n",
       "614764   Q3616234  Q14878358  UP_REGULATES_GuBP       NaN  Q14761970"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "edges_mike.tail()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "0a3fc565",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.00038121541341145835 for activesite.ndjson\n",
      "0.0034737348556518554 for anatomy.ndjson\n",
      "0.003675250212351481 for biologicalpathway.ndjson\n",
      "0.018889617919921876 for biologicalprocess.ndjson\n",
      "0.00014118353525797525 for bindingsite.ndjson\n",
      "0.0010748664538065592 for cellularcomponent.ndjson\n",
      "1.567999521891276e-05 for chemicalhazard.ndjson\n",
      "0.2745723843574524 for compound.ndjson\n",
      "0.0018672664960225423 for disease.ndjson\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-6-9ab4cc8702ae>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     48\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     49\u001b[0m \u001b[0mstart_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 50\u001b[0;31m \u001b[0mgene\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'files/gene.ndjson'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlines\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     51\u001b[0m \u001b[0mend_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     52\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mend_time\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mstart_time\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0;36m60\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"for gene.ndjson\"\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;31m# 3.5 min\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    197\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    198\u001b[0m                     \u001b[0mkwargs\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mnew_arg_name\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnew_arg_value\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 199\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    200\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    201\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mcast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mF\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/util/_decorators.py\u001b[0m in \u001b[0;36mwrapper\u001b[0;34m(*args, **kwargs)\u001b[0m\n\u001b[1;32m    297\u001b[0m                 )\n\u001b[1;32m    298\u001b[0m                 \u001b[0mwarnings\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwarn\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mFutureWarning\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mstacklevel\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mstacklevel\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 299\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    300\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    301\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mwrapper\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/io/json/_json.py\u001b[0m in \u001b[0;36mread_json\u001b[0;34m(path_or_buf, orient, typ, dtype, convert_axes, convert_dates, keep_default_dates, numpy, precise_float, date_unit, encoding, lines, chunksize, compression, nrows, storage_options)\u001b[0m\n\u001b[1;32m    561\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    562\u001b[0m     \u001b[0;32mwith\u001b[0m \u001b[0mjson_reader\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 563\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mjson_reader\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    564\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    565\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/io/json/_json.py\u001b[0m in \u001b[0;36mread\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    690\u001b[0m                 \u001b[0mdata\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mensure_str\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    691\u001b[0m                 \u001b[0mdata_lines\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"\\n\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 692\u001b[0;31m                 \u001b[0mobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_object_parser\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_combine_lines\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata_lines\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    693\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    694\u001b[0m             \u001b[0mobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_object_parser\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/io/json/_json.py\u001b[0m in \u001b[0;36m_get_object_parser\u001b[0;34m(self, json)\u001b[0m\n\u001b[1;32m    714\u001b[0m         \u001b[0mobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    715\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtyp\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"frame\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 716\u001b[0;31m             \u001b[0mobj\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mFrameParser\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mparse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    717\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    718\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mtyp\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"series\"\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0mobj\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/io/json/_json.py\u001b[0m in \u001b[0;36mparse\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    829\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    830\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 831\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_parse_no_numpy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    832\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    833\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/io/json/_json.py\u001b[0m in \u001b[0;36m_parse_no_numpy\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1077\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0morient\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"columns\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1078\u001b[0m             self.obj = DataFrame(\n\u001b[0;32m-> 1079\u001b[0;31m                 \u001b[0mloads\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mjson\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprecise_float\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mprecise_float\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1080\u001b[0m             )\n\u001b[1;32m   1081\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0morient\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m\"split\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "# Create time stamp\n",
    "timeStringNow = datetime.now().strftime(\"+%Y-%m-%dT00:00:00Z\") \n",
    "\n",
    "start_time = time.time()\n",
    "activesite = pd.read_json('files/activesite.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for activesite.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "anatomy = pd.read_json('files/anatomy.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for anatomy.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "biologicalpathway = pd.read_json('files/biologicalpathway.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for biologicalpathway.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "biologicalprocess = pd.read_json('files/biologicalprocess.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for biologicalprocess.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "bindingsite = pd.read_json('files/bindingsite.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for bindingsite.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "cellularcomponent = pd.read_json('files/cellularcomponent.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for cellularcomponent.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "chemicalhazard = pd.read_json('files/chemicalhazard.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for chemicalhazard.ndjson\") \n",
    "\n",
    "start_time = time.time()\n",
    "compound = pd.read_json('files/compound.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for compound.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "disease = pd.read_json('files/disease.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for disease.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "gene = pd.read_json('files/gene.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for gene.ndjson\") # 3.5 min\n",
    "\n",
    "start_time = time.time()\n",
    "medicalspecialty = pd.read_json('files/medicalspecialty.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for medicalspecialty.ndjson\") # 11 min\n",
    "\n",
    "start_time = time.time()\n",
    "molecularfunction = pd.read_json('files/molecularfunction.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for molecularfunction.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "protein = pd.read_json('files/protein.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for protein.ndjson\") # 17 min\n",
    "\n",
    "start_time = time.time()\n",
    "proteindomain = pd.read_json('files/proteindomain.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for proteindomain.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "proteinfamily = pd.read_json('files/proteinfamily.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for proteinfamily.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "sequencevariant = pd.read_json('files/sequencevariant.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for sequencevariant.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "structuralmotif = pd.read_json('files/structuralmotif.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for structuralmotif.ndjson\")\n",
    "\n",
    "start_time = time.time()\n",
    "symptom = pd.read_json('files/symptom.ndjson', lines=True) \n",
    "end_time = time.time()\n",
    "print((end_time - start_time)/60, \"for symptom.ndjson\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "e2f3c566",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Biological Process           24718\n",
       "Gene                         23737\n",
       "Protein Family               10406\n",
       "Compound                      8557\n",
       "Disease                       8302\n",
       "Protein Domain                5230\n",
       "Molecular Function            4397\n",
       "Biological Pathway            2461\n",
       "Sequence Variant              1835\n",
       "Cellular Component            1785\n",
       "Chemical Hazard                687\n",
       "Super-Secondary Structure      663\n",
       "Chemical Role                  499\n",
       "Symptom                        228\n",
       "Anatomical Structure           218\n",
       "Active Site                    114\n",
       "Structural Motif               109\n",
       "Binding Site                    69\n",
       "Medical Specialty               69\n",
       "Name: :LABEL, dtype: int64"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "nodes_mike[':LABEL'].value_counts() # 19 categories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "eb9c03b3",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Biological Process \t 3542\n",
      "Gene \t\t\t 89347\n",
      "Protein Family \t\t 2441\n",
      "Compound \t\t 18718\n",
      "Disease \t\t 335\n",
      "Protein Domain \t\t 1117\n",
      "Molecular Function \t 1301\n",
      "Biological Pathway \t 47\n",
      "Sequence Variant \t 71\n",
      "Cellular Component \t 510\n",
      "Chemical Role \t\t 0\n",
      "Anatomical Structure \t 309\n",
      "Active Site \t\t 9\n",
      "Structural Motif \t 35\n",
      "Binding Site \t\t 10\n",
      "Medical Specialty \t 63603\n",
      "\n",
      "Protein \t\t 63603\n"
     ]
    }
   ],
   "source": [
    "print( \"Biological Process \\t\", len(biologicalprocess)) \n",
    "print( \"Gene \\t\\t\\t\", len(gene)) \n",
    "print( \"Protein Family \\t\\t\", len(proteinfamily)) \n",
    "print( \"Compound \\t\\t\", len(compound))\n",
    "print( \"Disease \\t\\t\", len(disease))\n",
    "print( \"Protein Domain \\t\\t\", len(proteindomain))\n",
    "print( \"Molecular Function \\t\", len(molecularfunction))\n",
    "print( \"Biological Pathway \\t\", len(biologicalpathway))\n",
    "print( \"Sequence Variant \\t\", len(sequencevariant))\n",
    "print( \"Cellular Component \\t\", len(cellularcomponent))\n",
    "print( \"Chemical Hazard \\t\\t\", len(chemicalhazard))\n",
    "print( \"Anatomical Structure \\t\", len(anatomy))\n",
    "print( \"Active Site \\t\\t\", len(activesite))\n",
    "print( \"Structural Motif \\t\", len(structuralmotif))\n",
    "print( \"Binding Site \\t\\t\", len(bindingsite))\n",
    "print( \"Medical Specialty \\t\", len(medicalspecialty))\n",
    "print( \"\\nProtein \\t\\t\", len(protein)) \n",
    "\n",
    "#\n",
    "# Note that Mike has 24718 for biological process (ask Andra what he has)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "473f934a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import csv"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "id": "521e4c4c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "section took 0.009759986400604248 minutes\n",
      "                            :ID,:LABEL,name\n",
      "Q24721060 Active Site  InterPro Active Site\n",
      "Q24721301 Active Site  InterPro Active Site\n",
      "Q24721995 Active Site  InterPro Active Site\n",
      "Q24740369 Active Site  InterPro Active Site\n",
      "Q24770253 Active Site  InterPro Active Site\n",
      "                                                              :ID,:LABEL,name\n",
      "Q7365 Anatomical Structure                 contractile soft tissue of mammals\n",
      "Q9377 Anatomical Structure  internal organ in most animals, including vert...\n",
      "Q9609 Anatomical Structure  blood vessel that carry deoxygenated blood tow...\n",
      "Q9620 Anatomical Structure  enclosed, cable-like bundle of axons in the pe...\n",
      "Q9635 Anatomical Structure  first portion of the alimentary canal that rec...\n",
      "                                                              :ID,:LABEL,name\n",
      "Q1057   Biological Pathway  set of chemical reactions and pathways by whic...\n",
      "Q99974  Biological Pathway  symbiotic association between a fungus and the...\n",
      "Q134624 Biological Pathway  biological pollination process occurring in pl...\n",
      "Q173432 Biological Pathway                           ce of any sexual process\n",
      "Q179217 Biological Pathway  process in which the coagulation factors of th...\n",
      "                                                              :ID,:LABEL,name\n",
      "Q1057   Biological Process  set of chemical reactions and pathways by whic...\n",
      "Q99974  Biological Process  symbiotic association between a fungus and the...\n",
      "Q134624 Biological Process  biological pollination process occurring in pl...\n",
      "Q173432 Biological Process                           ce of any sexual process\n",
      "Q179217 Biological Process  process in which the coagulation factors of th...\n"
     ]
    }
   ],
   "source": [
    "timeStringNow = datetime.now().strftime(\"+%Y-%m-%dT00:00:00Z\") \n",
    "start_time = time.time()\n",
    "\n",
    "with open('activesite.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID', ':LABEL', 'name'])\n",
    "    for index, row in activesite.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Active Site#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Active Site#\", file = file)\n",
    "activesitefinal = pd.read_csv('activesite.csv', sep=\"#\")\n",
    "\n",
    "with open('anatomy.csv', 'w') as file:  \n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID', ':LABEL', 'name'])\n",
    "    for index, row in anatomy.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Anatomical Structure#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Anatomical Structure#\", file = file)\n",
    "anatomyfinal = pd.read_csv('anatomy.csv', sep=\"#\")\n",
    "\n",
    "with open('biologicalpathway.csv', 'w') as file:\n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID', ':LABEL', 'name'])\n",
    "    for index, row in biologicalprocess.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Biological Pathway#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Biological Pathway#\", file = file)\n",
    "biologicalpathwayfinal = pd.read_csv('biologicalpathway.csv', sep=\"#\")\n",
    "\n",
    "with open('biologicalprocess.csv', 'w') as file:\n",
    "    writer = csv.writer(file)\n",
    "    writer.writerow([':ID', ':LABEL', 'name'])\n",
    "    for index, row in biologicalprocess.iterrows():\n",
    "        if \"en\" in row[\"descriptions\"].keys():\n",
    "            print(row[\"id\"]+\"#Biological Process#\"+row[\"descriptions\"][\"en\"][\"value\"], file = file)\n",
    "        else:\n",
    "            print(row[\"id\"]+\"#Biological Process#\", file = file)\n",
    "biologicalprocessfinal = pd.read_csv('biologicalprocess.csv', sep=\"#\")\n",
    "\n",
    "end_time = time.time()\n",
    "print(\"section took\", (end_time - start_time)/60, \"minutes\") \n",
    "\n",
    "print(activesitefinal.head())\n",
    "print(anatomyfinal.head())\n",
    "print(biologicalpathwayfinal.head())\n",
    "print(biologicalprocessfinal.head())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "id": "d10b913b",
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "\"None of [Index([':ID', ':LABEL', 'name'], dtype='object')] are in the [columns]\"",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-58-424530469151>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      1\u001b[0m \u001b[0;31m# make sure same number of columns (1:3)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 2\u001b[0;31m \u001b[0mactivesitefinal\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mactivesitefinal\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m':ID'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m':LABEL'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      3\u001b[0m \u001b[0manatomyfinal\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0manatomyfinal\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m':ID'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m':LABEL'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mbiologicalpathwayfinal\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbiologicalpathwayfinal\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m':ID'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m':LABEL'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m \u001b[0mbiologicalprocessfinal\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbiologicalprocessfinal\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m':ID'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m':LABEL'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'name'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3028\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_iterator\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3029\u001b[0m                 \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3030\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mloc\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_listlike_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mraise_missing\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mTrue\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3031\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3032\u001b[0m         \u001b[0;31m# take() does not accept boolean indexers\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_get_listlike_indexer\u001b[0;34m(self, key, axis, raise_missing)\u001b[0m\n\u001b[1;32m   1264\u001b[0m             \u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnew_indexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0max\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_reindex_non_unique\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1265\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1266\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_validate_read_indexer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mraise_missing\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mraise_missing\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1267\u001b[0m         \u001b[0;32mreturn\u001b[0m \u001b[0mkeyarr\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindexer\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1268\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/opt/anaconda3/lib/python3.8/site-packages/pandas/core/indexing.py\u001b[0m in \u001b[0;36m_validate_read_indexer\u001b[0;34m(self, key, indexer, axis, raise_missing)\u001b[0m\n\u001b[1;32m   1306\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mmissing\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1307\u001b[0m                 \u001b[0maxis_name\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis_name\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1308\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34mf\"None of [{key}] are in the [{axis_name}]\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1309\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1310\u001b[0m             \u001b[0max\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mobj\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_axis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: \"None of [Index([':ID', ':LABEL', 'name'], dtype='object')] are in the [columns]\""
     ]
    }
   ],
   "source": [
    "# make sure same number of columns (1:3)\n",
    "activesitefinal = activesitefinal[[':ID', ':LABEL', 'name']]\n",
    "anatomyfinal = anatomyfinal[[':ID', ':LABEL', 'name']]\n",
    "biologicalpathwayfinal = biologicalpathwayfinal[[':ID', ':LABEL', 'name']]\n",
    "biologicalprocessfinal = biologicalprocessfinal[[':ID', ':LABEL', 'name']]\n",
    "\n",
    "\n",
    "nodes = pd.concat([activesitefinal, anatomyfinal, biologicalpathwayfinal, biologicalprocessfinal], axis=0)\n",
    "nodes.to_csv('nodes.csv')\n",
    "\n",
    "print(nodes.head())\n",
    "print(nodes.tail())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f7bf8e46",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
